alethzero/plugins/whisper/Whisper.cpp

```diff
@@ -175,70 +207,87 @@ void Whisper::on_newIdentity_triggered()
 	refreshWhisper();
 }
 
-void Whisper::on_post_clicked()
+void Whisper::onPostClicked()
 {
 	QString const qsTopic = m_ui->shhTopic->currentText();
 	noteTopic(qsTopic);
 
-	QString const qsDest = m_ui->shhTo->currentText();
-	Public dest = stringToPublic(qsDest);
+	QString const strDest = m_ui->shhTo->currentText();
+	Public dest = stringToPublic(strDest);
 	if (dest)
-		noteDestination(qsDest);
-	else if (!qsDest.isEmpty())
-		return; // don't allow unencrypted messages, unless it was explicitly intended
+	{
+		if (m_ui->shhTo->findText(strDest) < 0)
+			m_ui->shhTo->addItem(strDest);
+	}
+	else if (!strDest.isEmpty())
+	{
+		// don't allow unencrypted messages, unless it was explicitly intended
+		QMessageBox box(QMessageBox::Warning, "Warning", "Invalid destination address!");
+		box.setInformativeText("Destination address must be either valid or blank.");
+		box.exec();
+		return;
+	}
 
-	string text = m_ui->shhData->toPlainText().toStdString();
-	if (text.empty())
+	int const msgSizeLimit = 1024;
```

Comment:
Its reasonable to have a limit although I'm not sure 1024 is sufficient. Can/should this be declared elsewhere?  Someone could submit a larger message to rpc...